### PR TITLE
When building a vector store with use_async=True it doesn't set a TTL on records

### PR DESCRIPTION
**Pull request by mvaquero-atlasti**
# Description

Sorry if something is missing, this is my first PR and it was a really small change.
When using a vector store and passing in the parameter use_async in the constructor, the additional kwargs are not passed when the add method is called.
For example:
 ```python
VectorStoreIndex(
        nodes,
        storage_context=storage_context,
        ttl=3600,
        use_async=True,
)
```
This creates the embeddings for each node and inserts them into the vector store, but doesn't set a TTL if use_async=True. However, it does set a TTL if use_async=False.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-core/llama_index/core/vector_stores/types.py [hunk 1]
```diff
@@ -363,7 +363,7 @@ async def async_add(
         NOTE: this is not implemented for all vector stores. If not implemented,
         it will just call add synchronously.
         """
-        return self.add(nodes)
+        return self.add(nodes, **kwargs)
```
[the remark this answers is not in the document]
Comment (logan-markewich) [reply llama-index-core/llama_index/core/vector_stores/types.py:366]: (At least in the base class, I'm not sure if you tested this or what class you tested with)

[the remark this answers is not in the document]
Comment (mvaquero-atlasti) [reply llama-index-core/llama_index/core/vector_stores/types.py:366]: thanks for taking a look @logan-markewich! It does work, at least with the Redis vector store.
This is the code I've tried it with:
```python
vector_store = RedisVectorStore(redis_client=redis_client, schema=index_schema)
storage_context = StorageContext.from_defaults(vector_store)
return VectorStoreIndex(
    nodes,
    storage_context=storage_context,
    ttl=3600,
    use_async=True,
)
```

I see the RedisVectorStore class does accept kwargs in `add` here:
- https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/vector_stores/llama-index-vector-stores-redis/llama_index/vector_stores/redis/base.py#L220

That's why it works with Redis. Maybe we also need to add the param to the `self.add` in the base class (and some other places?)

[the remark this answers is not in the document]
Comment (logan-markewich) [reply llama-index-core/llama_index/core/vector_stores/types.py:366]: huh weird, seems like only the base class is missing this lol. Every other vector store has kwargs

[the remark this answers is not in the document]
Comment (mvaquero-atlasti) [reply llama-index-core/llama_index/core/vector_stores/types.py:367]: Thanks for adding it to base as well! Hope this makes it into the next version :)

**Pull request merged**